src/core/server/elasticsearch/elasticsearch_service.test.ts

```diff
@@ -0,0 +1,158 @@
+    const startContract = await elasticsearchService.start();
+
+    await expect(startContract.legacy.config$.pipe(first()).toPromise()).resolves
+      .toMatchInlineSnapshot(`
+ElasticsearchConfig {
+  "apiVersion": "master",
+  "customHeaders": Object {},
+  "healthCheckDelay": "PT2.5S",
+  "hosts": Array [
+    "http://1.2.3.4",
+  ],
+  "logQueries": false,
+  "password": undefined,
+  "pingTimeout": "PT30S",
+  "requestHeadersWhitelist": Array [
+    "authorization",
+  ],
+  "requestTimeout": "PT30S",
+  "shardTimeout": "PT30S",
+  "sniffInterval": false,
+  "sniffOnConnectionFault": false,
+  "sniffOnStart": false,
+  "ssl": Object {
+    "alwaysPresentCertificate": true,
+    "certificateAuthorities": undefined,
+    "verificationMode": "full",
+  },
+  "username": "jest",
+}
+`);
+  });
+
+  test('returns data and admin client observables as a part of the contract', async () => {
+    const mockAdminClusterClientInstance = { close: jest.fn() };
+    const mockDataClusterClientInstance = { close: jest.fn() };
+    MockClusterClient.mockImplementationOnce(
+      () => mockAdminClusterClientInstance
+    ).mockImplementationOnce(() => mockDataClusterClientInstance);
+
+    const startContract = await elasticsearchService.start();
+
+    const [esConfig, adminClient, dataClient] = await combineLatest(
+      startContract.legacy.config$,
+      startContract.adminClient$,
+      startContract.dataClient$
+    )
+      .pipe(first())
+      .toPromise();
+
+    expect(adminClient).toBe(mockAdminClusterClientInstance);
+    expect(dataClient).toBe(mockDataClusterClientInstance);
+
+    expect(MockClusterClient).toHaveBeenCalledTimes(2);
+    expect(MockClusterClient).toHaveBeenNthCalledWith(
+      1,
+      esConfig,
+      expect.objectContaining({ context: ['elasticsearch', 'admin'] })
+    );
+    expect(MockClusterClient).toHaveBeenNthCalledWith(
+      2,
+      esConfig,
+      expect.objectContaining({ context: ['elasticsearch', 'data'] })
+    );
+
+    expect(mockAdminClusterClientInstance.close).not.toHaveBeenCalled();
+    expect(mockDataClusterClientInstance.close).not.toHaveBeenCalled();
+  });
+
+  test('returns `createClient` as a part of the contract', async () => {
+    const startContract = await elasticsearchService.start();
+
+    const mockClusterClientInstance = { close: jest.fn() };
+    MockClusterClient.mockImplementation(() => mockClusterClientInstance);
+    MockClusterClient.mockClear();
+
+    const mockConfig = { logQueries: true };
+    const clusterClient = startContract.createClient('some-custom-type', mockConfig as any);
+
+    expect(clusterClient).toBe(mockClusterClientInstance);
+
+    expect(MockClusterClient).toHaveBeenCalledWith(
+      mockConfig,
+      expect.objectContaining({ context: ['elasticsearch', 'some-custom-type'] })
+    );
+  });
+});
+
+describe('#stop', () => {
+  test('stops both admin and data clients', async () => {
```

Comment:
I don't think we ever want to re-start any of the core services after we stopped them. At this point `stop` is more like `cleanup-before-shutdown` rather than `reset`. I'd rather add such tests for our services when we really need them.